### PR TITLE
Feature/email client compose

### PR DESCRIPTION
**Pull request by daniel-samson**
## Description

This changes adds a new view called ComposeView. It inherits its behavior from the EditView. It has its own metadata file modules/Emails/metadata/composeviewdefs.php.

screenshot_20170323_103218

### Requirements
* User can compose an email in a modal dialog on the list view
* User can create html emails with attachments

In order to comply to these requirements, I have added some new features for developers. There is a new jquery plugin called message box, which enables the developer to dynamically create modal dialogs in suitecrm using javascript. This feature is application wide.

#### Typical Usage
```
var mb = messageBox();
mb.setTitle('example');
mb.setBody('Hello World');
mb.show();

mb.on('ok', function() {
    "use strict";
    // do somthing
    mb.remove();
});

mb.on('cancel', function() {

"use strict";

    // do something
    mb.remove();
})
```
Developers can mix message box with the modal plugin to create a custom user experience.

Developers can extend the email functionality by editing the composeviewdefs.php. They can add buttons just like the would in the edit view. This will add a button to the compose toolbar. They can then include javascript in the same manner so that they can add the client side logic.

```
$viewdefs['Emails']['ComposeView'] = array(
    'templateMeta' => array(
        'form' => array(
            'footerTpl' => 'modules/Emails/include/ComposeView/ComposeViewToolbar.tpl',
            'buttons' => array(
                array('customCode' => '<button class="btn btn-send-email" title="{$MOD.LBL_SEND_BUTTON_TITLE}"><span class="glyphicon glyphicon-send"></span></button>'),
                array('customCode' => '<button class="vertical-seperator"></button>'),
                array('customCode' => '<button class="btn btn-attach-file" title="{$MOD.LBL_ATTACH_FILES}"><span class="glyphicon glyphicon-paperclip"></span></button>'),
                array('customCode' => '<button class="btn btn-attach-document" title="{$MOD.LBL_ATTACH_DOCUMENTS}"><span class="glyphicon"><img src="themes/SuiteP/images/sidebar/modules/Documents.svg"></span></button>'),
                array('customCode' => '<button class="vertical-seperator"></button>'),
                array('customCode' => '<button class="btn btn-save-draft" title="{$MOD.LBL_SAVE_AS_DRAFT_BUTTON_TITLE}"><span class="glyphicon glyphicon-floppy-save"></span></button>'),
                array('customCode' => '<button class="btn btn-disregard-draft" title="{$MOD.LBL_DISREGARD_DRAFT_BUTTON_TITLE}"><span class="glyphicon glyphicon-trash"></span></button>'),
                // Add your button here
            )
        ),
        'includes' => array(
            array(
                'file' => 'modules/Emails/include/ComposeView/EmailsComposeView.js',
            ),
            array(
                'file' => 'include/javascript/mozaik/vendor/tinymce/tinymce/tinymce.min.js'
            )
                // Add your js here
        ),
    ),
```
This can also be extended using the extensions framework in the custom folder.

3rd party plugins can utilise the validation and other methods in the EmailComposeView jquery plugin. 
 They can also extend this functionality.

In this pull request, I have added a new class **UploadMultipleFiles**, which add support to SuiteCRM so that can handle when multiple files are send by a single field. This is handled in the **Emails->handleMultipleFileAttachments()** method.

I have tried to make the user experience for Emails attachments and document consistent. Eg A user can change a attachment by clicking on the dark purple part of the attachment. The can remove an attachment by clicking the "X".

Email attachments now shows an icon based on the file type:

screenshot_20170321_163013

The attach documents also utilises the icon from the documents module.

screenshot_20170323_115843

## Motivation and Context

* Makes it more compliant with our design
* Uses the same functionality that is already in SuiteCRM
* It is easy to extend or change by developers
* Creates a consistent user experience

## How To Test This

* Repair and rebuild
* Repair JS
* Repair JS Groupings
* Rebuild Sass
* Clear Browser Cache
* Compose an email
* Test Validation
* Test Document Attachments
* Test Sending Emails
* Test Html features

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [x] My code follows the code style of this project found [here](https://suitecrm.com/wiki/index.php/Coding_Standards).
- [x] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM) guidelines.

### CODE REVIEW

File: modules/Emails/language/en_us.lang.php [hunk 1]
```diff
@@ -340,6 +345,11 @@
     'LBL_ADDRESS_BOOK_SEARCH_HELP' => 'Enter an email address, First Name, Last Name or Account Name to find recipients.',
     'LBL_TEST_SETTINGS'     => 'Test Settings',
     'LBL_EMPTY_EMAIL_BODY'  => '<p><span style="color: #888888;"><em>This Message Has No Content</em></span></p>',
+    'LBL_HAS_EMPTY_EMAIL_SUBJECT' => 'Please specify the subject',
+    'LBL_HAS_EMPTY_EMAIL_BODY' => 'Please specify the your message in the body',
```
Comment (horus68) [inline modules/Emails/language/en_us.lang.php]: Consider to remove THE
to
`Please specify your message in the body`

Comment (daniel-samson) [inline modules/Emails/language/en_us.lang.php]: Ah yes...well spotted. I will remove it. :+1:

Comment (daniel-samson) [inline modules/Emails/language/en_us.lang.php]: I have removed the word "the". Again thanks for telling me.

**Comment by daniel-samson** [comment]
I have amended my pull request to meet your requests.

**Pull request merged**